Shuttle-back/src/main/java/com/shuttle/driver/DriverController.java

```diff
@@ -1,36 +1,55 @@
 package com.shuttle.driver;
 
+import com.fasterxml.jackson.annotation.JsonFormat;
 import com.shuttle.common.ListDTO;
 import com.shuttle.driver.dto.DriverDTO;
 import com.shuttle.driver.dto.DriverDataPageDTO;
 import com.shuttle.driver.dto.DriverDocumentDTO;
+import com.shuttle.ride.IRideRepository;
+import com.shuttle.ride.Ride;
 import com.shuttle.ride.dto.RideDTO;
 import com.shuttle.vehicle.VehicleDTO;
 import com.shuttle.workhours.WorkHours;
+import jakarta.annotation.security.PermitAll;
 import jakarta.websocket.server.PathParam;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.format.annotation.DateTimeFormat;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
 import java.util.ArrayList;
 import java.util.List;
 
+
 @CrossOrigin
 @RestController
 public class DriverController {
-	private IDriverService driverService;
-	
-	@Autowired
-	public DriverController(IDriverService driverService) {
-		this.driverService = driverService;
-	}
+
+    @Autowired
+    private IDriverService driverService;
+    @Autowired
+    public IRideRepository rideRepository;
+
+    @Autowired
+    public DriverController(IDriverService driverService) {
```

Comment:
If you're injecting an `IDriverService` through an attribute injector (see above), then you don't need to inject it again with a ctor injector.